kedro-telemetry/tests/test_plugin.py

```diff
@@ -128,19 +127,6 @@ def test_before_command_run_no_consent_given(self, mocker, fake_metadata):
 
         mocked_heap_call.assert_not_called()
 
-    def test_before_command_run_socket_timeout(self, mocker, fake_metadata):
```

Comment:
Why did you remove this test?

Reply:
I noticed now you removed the timeout logic.

Reply:
> This implementation looks good to me 🙂
> 
> On the point of grouping all "anonymous" events together: I think we should keep the same behaviour as before where if we can't determine the username we don't send any data at all. You've already switched to `getpass.getuser()` which is more robust then `os.getlogin()` afaik. But it would be good to hear what @yetudada thinks about this.

So I think following standup, it should be that if we can't determine the `username` they should get the unique Heap ID.

Reply:
Perfect, I made this change in [7181e4d](https://github.com/kedro-org/kedro-plugins/pull/58/commits/7181e4dcbdeba368b6e2b8cbbac0049a138d04ef).

As I understand it from the heap docs, if the request is made with no `identity` field, heap automatically assigns the unique ID. In the case that no username can be found, the `identity` field is removed from the request.